null reference exception if specified "scope" in deploymentconfig.json

If "scope" is set in deploymentconfig.json the error below will happend.

`Resolve-TemplateDeploymentScope: /home/runner/work/_actions/climpr/deploy-bicep/v1/src/Resolve-DeploymentConfig.ps1:84
Line |
  84 |  …           = Resolve-TemplateDeploymentScope -DeploymentFilePath $depl …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'Path' because it is null.
Error: Process completed with exit code 1.`

This happens becuase `$referenceString` will not be set in function `Resolve-TemplateDeploymentScope`:

 ` if ($DeploymentConfig.scope) {
        Write-Debug "[Resolve-TemplateDeploymentScope()] TargetScope determined by scope property in deploymentconfig.json file"
        $targetScope = $DeploymentConfig.scope
    }
    elseif ($deploymentFile.Extension -eq ".bicep") {
        $referenceString = $deploymentFile.Name
    }
    elseif ($deploymentFile.Extension -eq ".bicepparam") {
        $referenceString = Resolve-ParameterFileTarget -Path $DeploymentFilePath
    }
    else {
        throw "Deployment file extension not supported. Only .bicep and .bicepparam is supported. Input deployment file extension: '$($deploymentFile.Extension)'"
    }`

And later in the function $referenceString will be called upon, which is now not set.
`$content = Get-Content -Path $referenceString -Raw`

The workaroud here is simply to not specify "scope" in the (env.)deploymentconfig.json(c) file. and instead rely on targetScope in your main.bicep file.

--- a/src/support-functions.psm1
+++ b/src/support-functions.psm1
@@ -149,11 +149,7 @@ function Resolve-TemplateDeploymentScope {
     $targetScope = ""
     $deploymentFile = Get-Item -Path $DeploymentFilePath
     
-    if ($DeploymentConfig.scope) {
-        Write-Debug "[Resolve-TemplateDeploymentScope()] TargetScope determined by scope property in deploymentconfig.json file"
-        $targetScope = $DeploymentConfig.scope
-    }
-    elseif ($deploymentFile.Extension -eq ".bicep") {
+    if ($deploymentFile.Extension -eq ".bicep") {
         $referenceString = $deploymentFile.Name
     }
     elseif ($deploymentFile.Extension -eq ".bicepparam") {

--- a/src/tests/Resolve-TemplateDeploymentScope.tests.ps1
+++ b/src/tests/Resolve-TemplateDeploymentScope.tests.ps1
@@ -68,4 +68,21 @@ Describe "Resolve-TemplateDeploymentScope.ps1" {
             $templateDeploymentScope | Should -Be 'subscription'
         }
     }
+
+    Context "When scope is defined in deploymentcondif.jsonc" {
+        BeforeAll {
+            $script:param = @{
+                DeploymentFilePath = "$mockDirectory/deployments/deployment/comments/targetScopeCommented.bicepparam"
+                DeploymentConfig   = @{
+                    'managementGroupId' = 'mockMgmtGroupId'
+                    'scope'             = 'subscription'
+                }
+            }
+            $script:templateDeploymentScope = Resolve-TemplateDeploymentScope @param
+        }
+
+        It "Should resolve DeploymentScope to be [subscription]" {
+            $templateDeploymentScope | Should -Be 'subscription'
+        }
+    }
 }

--- a/src/tests/Resolve-TemplateDeploymentScope.tests.ps1
+++ b/src/tests/Resolve-TemplateDeploymentScope.tests.ps1
@@ -69,7 +69,7 @@ Describe "Resolve-TemplateDeploymentScope.ps1" {
         }
     }
 
-    Context "When scope is defined in deploymentcondif.jsonc" {
+    Context "When scope is defined in deploymentconfig.jsonc" {
         BeforeAll {
             $script:param = @{
                 DeploymentFilePath = "$mockDirectory/deployments/deployment/comments/targetScopeCommented.bicepparam"

